packages/access-client/src/types.ts

```diff
@@ -350,3 +351,9 @@ export type InvokeAndExecute = <
     import('./types').Service
   >
 >
+
+export type AuthorizationWaiterOpts<T> = { signal?: AbortSignal } & T
```

Comment:
Does this need to be generic over `T`? It wasn't before.

```suggestion
export type AuthorizationWaiterOpts = { signal?: AbortSignal }
```

If someone wants to 'mixin' `T`, they can do `AuthorizationWaiterOpts & T`

Reply:
I also don't really like moving it to this file at the top of th eproject, because now this type
* is far away from the thing that uses it
* is exported from `@web3-storage/access` and not just `@web3-storage/access/agent`
* doesn't have any jsdoc, so the fact that it's far away from its usage means its less understandable and more likely to be used incorrectly